rpc/tls.go

```diff
@@ -96,12 +96,29 @@ func LoadTLSConfig(certPEM, keyPEM, caPEM []byte) (*TLSConfig, error) {
 	return &TLSConfig{
 		config: &tls.Config{
 			Certificates: []tls.Certificate{cert},
-			ClientAuth:   tls.RequireAndVerifyClientCert,
-			RootCAs:      certPool,
-			ClientCAs:    certPool,
-
-			// TODO(jqmp): Set CipherSuites?
-			// TODO(jqmp): Set MinVersion?
+			// TODO(marc): clients are bad about this. We should switch to
+			// tls.RequireAndVerifyClientCert once client certs are properly set.
+			ClientAuth: tls.VerifyClientCertIfGiven,
+			RootCAs:    certPool,
+			ClientCAs:  certPool,
+
+			// Ciphers from http://golang.org/src/crypto/tls/cipher_suites.go
+			// but with RC4 and 3DES removed.
+			CipherSuites: []uint16{
+				tls.TLS_RSA_WITH_AES_128_CBC_SHA,
+				tls.TLS_RSA_WITH_AES_256_CBC_SHA,
+				tls.TLS_ECDHE_ECDSA_WITH_AES_128_CBC_SHA,
+				tls.TLS_ECDHE_ECDSA_WITH_AES_256_CBC_SHA,
+				tls.TLS_ECDHE_RSA_WITH_AES_128_CBC_SHA,
+				tls.TLS_ECDHE_RSA_WITH_AES_256_CBC_SHA,
+				tls.TLS_ECDHE_RSA_WITH_AES_128_GCM_SHA256,
+				tls.TLS_ECDHE_ECDSA_WITH_AES_128_GCM_SHA256},
+
+			// Use only TLS v1.2.
+			MinVersion: tls.VersionTLS12,
+
+			// Should we disable session resumption? This may break forward secrecy.
```

Comment:
Seems reasonable. Our connections tend to be reliably long-lived and are not terribly latency-sensitive so session resumption is not very important for us. OTOH, I'm wary of non-default (and therefore less tested) code paths here; this one in particular has had a security bug: https://groups.google.com/forum/#%21msg/golang-nuts/eeOHNw_shwU/OHALUmroA5kJ